src/v/redpanda/admin/server.cc

```diff
@@ -1698,6 +1698,28 @@ void config_multi_property_validation(
         errors[ss::sstring(name)] = ssx::sformat(
           "{} requires schema_registry to be enabled in redpanda.yaml", name);
     }
+
+    // Cloud storage cache size validation
```

Comment:
> To prevent the invalid configuration of `cloud_storage_cache_size`
and `cloud_storage_cache_size_percent` both equal to 0, add a new check
to the `admin` server validation post `patch_cluster_config_handler()` call.

* can you add to the commit message details about why the validation is done here as opposed to the cache itself?
* even if the enforcement is done here, i think it makes sense to have the logic live along side the cache itself?

Reply:
I agree, having a `cache::validate_config_update(...)` method would make sense.

Reply:
> can you add to the commit message details about why the validation is done here as opposed to the cache itself?

Done

> even if the enforcement is done here, i think it makes sense to have the logic live along side the cache itself?

Moved logic to a new static function, `cloud_storage::cache::validate_cache_config()`. I agree, this is much nicer :+1: